src/Parsers/Access/ASTAuthenticationData.cpp

```diff
@@ -131,7 +131,7 @@ void ASTAuthenticationData::formatImpl(const FormatSettings & settings, FormatSt
 
     if (password && !settings.show_secrets)
     {
-        prefix = "";
+        prefix += " `[HIDDEN]'";
```

Comment:
The `prefix` variable is not a semantically correct place to store that HIDDEN placeholder.

In line 56 prefix is defined as `String prefix; /// "BY" or "SERVER" or "REALM"`. Moreover, it is later output with the keyword hiliting in line  149 but HIDDEN is not a keyword, for sure.

You'll probably need to understand this file a bit more thoroughly to put the HIDDEN placeholder in the correct place. I can't point you to it as I myself don't know for sure :)